packages/caip/src/assetId/assetId.ts

```diff
@@ -124,26 +130,24 @@ export type FromAssetId = (assetId: AssetId) => FromAssetIdReturn
 
 export const fromAssetId: FromAssetId = (assetId) => {
   if (!isAssetId(assetId)) throw new Error(`fromAssetId: invalid AssetId: ${assetId}`)
-  const matches = parseAssetIdRegExp.exec(assetId)?.groups
+  const matches = parseAssetIdRegExp.exec(assetId)
   if (!matches) throw new Error(`fromAssetId: could not parse AssetId: ${assetId}`)
 
-  // These should never throw because isAssetId() would have already caught it, but they help with type inference
-  assertIsChainNamespace(matches.chainNamespace)
-  assertIsChainReference(matches.chainReference)
-  assertIsAssetNamespace(matches.assetNamespace)
-
-  const chainNamespace = matches.chainNamespace
-  const chainReference = matches.chainReference
-  const assetNamespace = matches.assetNamespace
-
+  const chainNamespace = matches[ASSET_ID_CAPTURE_GROUPS.CHAIN_NAMESPACE_CAPTURE_GROUP]
```

Comment:
@pastaghost this might be helpful

named captured groups were just a nice developer experience improvement, and not really essential for this regex to perform its intended purpose

we can simply destructure the same data out as an array 

`const [chainNamespace, chainReference, assetNamespace, assetReference] = matches`

your regex behaves exactly the same

https://regex101.com/r/oGdXUs/1